src/main/java/net/minecraftforge/registries/RegistryManager.java

```diff
@@ -133,9 +141,25 @@ public String getName()
             this.synced.add(name);
         for (ResourceLocation legacyName : builder.getLegacyNames())
             addLegacyName(legacyName, name);
+        if (registerToRoot && this == RegistryManager.ACTIVE && builder.getHasWrapper() && !Registry.REGISTRY.containsKey(reg.getRegistryName()) && Registry.REGISTRY instanceof WritableRegistry<?> rootRegistry)
+            registerRegistry(rootRegistry, reg);
         return getRegistry(name);
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> void registerRegistry(WritableRegistry<T> rootRegistry, ForgeRegistry<?> forgeReg) {
```

Comment:
Formatting: braces on newlines.
```suggestion
    private <T> void registerRegistry(WritableRegistry<T> rootRegistry, ForgeRegistry<?> forgeReg)
    {
```